lib/server.rb

```diff
@@ -328,7 +330,13 @@ def diff_to_object(diff)
   end
 
   get "/:id" do
-    @description = SystemDescription.load(params[:id], settings.system_description_store)
+    begin
+      @description = SystemDescription.load(params[:id], settings.system_description_store)
+    rescue Machinery::Errors::SystemDescriptionNotFound => e
+      session[:error] = e.to_s
+      status 404
```

Comment:
There is no point in setting the status, when the response is a redirect, or is there?